storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncProcessEvent.java

```diff
@@ -0,0 +1,428 @@
+            } else {
+                file.getParentFile().mkdirs();
+            }
+        }
+        FileWriter writer = new FileWriter(file);
+        Yaml yaml = new Yaml();
+        try {
+            yaml.dump(data, writer);
+        }finally {
+            writer.close();
+        }
+    }
+
+    /**
+     * Create a symlink from workder directory to its port artifacts directory
+     * 
+     * @param conf
+     * @param stormId
+     * @param port
+     * @param workerId
+     */
+    protected void createArtifactsLink(Map conf, String stormId, Long port, String workerId) throws IOException {
+        String workerDir = ConfigUtils.workerRoot(conf, workerId);
+        String topoDir = ConfigUtils.workerArtifactsRoot(conf, stormId);
+        if (Utils.checkFileExists(workerDir)) {
+            Utils.createSymlink(workerDir, topoDir, "artifacts", String.valueOf(port));
+        }
+    }
+
+    /**
+     * Create symlinks in worker launch directory for all blobs
+     * 
+     * @param conf
+     * @param stormId
+     * @param workerId
+     * @throws IOException
+     */
+    protected void createBlobstoreLinks(Map conf, String stormId, String workerId) throws IOException {
+        String stormRoot = ConfigUtils.supervisorStormDistRoot(conf, stormId);
+        Map stormConf = ConfigUtils.readSupervisorStormConf(conf, stormId);
+        String workerRoot = ConfigUtils.workerRoot(conf, workerId);
+        Map<String, Map<String, Object>> blobstoreMap = (Map<String, Map<String, Object>>) stormConf.get(Config.TOPOLOGY_BLOBSTORE_MAP);
+        List<String> blobFileNames = new ArrayList<>();
+        if (blobstoreMap != null) {
+            for (Map.Entry<String, Map<String, Object>> entry : blobstoreMap.entrySet()) {
+                String key = entry.getKey();
+                Map<String, Object> blobInfo = entry.getValue();
+                String ret = null;
+                if (blobInfo != null && blobInfo.containsKey("localname")) {
+                    ret = (String) blobInfo.get("localname");
+                } else {
+                    ret = key;
+                }
+                blobFileNames.add(ret);
+            }
+        }
+        List<String> resourceFileNames = new ArrayList<>();
+        resourceFileNames.add(ConfigUtils.RESOURCES_SUBDIR);
+        resourceFileNames.addAll(blobFileNames);
+        LOG.info("Creating symlinks for worker-id: {} storm-id: {} for files({}): {}", workerId, stormId, resourceFileNames.size(), resourceFileNames);
+        Utils.createSymlink(workerRoot, stormRoot, ConfigUtils.RESOURCES_SUBDIR);
+        for (String fileName : blobFileNames) {
+            Utils.createSymlink(workerRoot, stormRoot, fileName, fileName);
+        }
+    }
+
+    public void shutWorker(SupervisorData supervisorData, IWorkerManager workerManager, String workerId) throws IOException, InterruptedException{
```

Comment:
can we name this function to something other than shutWorker since there is already a shutdownWorker function and the similarities the names may cause confusion